src/librustc/ty/constness.rs

```diff
@@ -63,13 +103,81 @@ impl<'tcx> TyCtxt<'tcx> {
 
 
 pub fn provide(providers: &mut Providers<'_>) {
-    /// only checks whether the function has a `const` modifier
+    /// Const evaluability whitelist is here to check evaluability at the
+    /// top level beforehand.
+    fn is_const_intrinsic(tcx: TyCtxt<'_>, def_id: DefId) -> Option<bool> {
+        match tcx.fn_sig(def_id).abi() {
+            Abi::RustIntrinsic |
+            Abi::PlatformIntrinsic => {
+                match tcx.item_name(def_id) {
+                    // Keep this list in the same order as the match patterns in
+                    // `librustc_mir/interpret/intrinsics.rs`
+
+                    // This whitelist is a list of intrinsics that have a miri-engine implementation
+                    // and can thus be called when enabling enough feature gates. The similar
+                    // whitelist in `is_intrinsic_min_const_fn` (in this file), exists for allowing
+                    // the intrinsics to be transitively called by stable const fns.
```

Comment:
Why "transitively called"? Surely stable const fns can use a `const` item that does non-min-const-fn things?

Please also add as sentence saying that stability for other const contexts (such as const and static initializers) is controlled by the usual stability attributes at the intrinsic declaration.

Reply:
No, stable const fns can only do min const fn things, that's the entire point of min const fn. Although we watered that down with e.g. `if` so we can make more things const fn without yet stabilizing the feature

Reply:
But I referred to a `const` *item*.

Reply:
oh yea true